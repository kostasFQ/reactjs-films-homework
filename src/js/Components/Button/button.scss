.container {
  border: solid 2px gray;
  border-radius: 5px;
  padding: 1em 1.5em;
  cursor: pointer;
  transition: 0.5s;
  background: none;
  color: white;

  &:hover {
    background-color: rgb(9, 173, 227);
  }
}
