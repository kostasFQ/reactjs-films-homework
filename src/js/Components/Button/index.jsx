import React from 'react';
import PropTypes from 'prop-types';
import s from './button.scss';

const Button = (props) => {
  const { name, action } = props;
  return (
    <button className={s.container} onClick={action} type="button">
      {name}
    </button>
  );
};

export default Button;

Button.propTypes = {
  action: PropTypes.func,
  name: PropTypes.string.isRequired,
};
Button.defaultProps = {
  action: null,
};
