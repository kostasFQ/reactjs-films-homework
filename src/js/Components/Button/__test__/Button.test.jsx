import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

test('test component', () => {
  const render = renderer.create(
    <Button name="button" />,
  );
  const component = render.toJSON();

  expect(component).toMatchSnapshot();
  expect(component.type).toBe('button');
});
