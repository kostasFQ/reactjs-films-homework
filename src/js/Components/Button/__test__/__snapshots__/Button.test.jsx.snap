// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component 1`] = `
<button
  className="container"
  onClick={null}
  type="button"
>
  button
</button>
`;
